### PR TITLE
Add test for sequence point issue with local function and conditional

### PR DESCRIPTION
**Pull request by jcouv**
This is a test-only change to verify the sequence points in a method with a conditional followed by a local function. I also verified the scenario manually (in 15.4 preview 3) and it behaved as expected.

**Bugs this fixes:**
Closes https://github.com/dotnet/roslyn/issues/21324

@dotnet/roslyn-compiler for review. Thanks
FYI @tmat

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs [hunk 1]
```diff
@@ -2078,6 +2078,113 @@ .maxstack  2
...
+  // sequence point: System.Console.Write(1);
+  IL_000f:  ldc.i4.1
+  IL_0010:  call       ""void System.Console.Write(int)""
+  IL_0015:  nop
+  // sequence point: }
+  IL_0016:  ret
+}
+", sequencePoints: "C.M", source: source);
+
+            v.VerifyPdb(@"<symbols>
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: Any reason for extra PDB validation? The sequence points are already validated above.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: It was requested in a previous PR (as shown in test above) so I did the same. I don't feel strongly about it though.

Comment (tmat) [reply src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: If the purpose is to test the sequence points then that's already covered by the first call and the second is just adding clutter, imo.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: Removed the extra checks in last commit. Thanks

**Comment by tmat** [comment]
👍

**Comment by jcouv** [comment]
Thanks
@dotnet/roslyn-compiler for review (small test-only change).

**Comment by jcouv** [comment]
@jaredpar for ask-mode approval. Thanks

**Comment by gafter** [comment]
@jcouv You don't need approval, as there are no product changes in this PR.

**Comment by jaredpar** [comment]
Approved for ask mode.

**Pull request merged**